fews_3di/utils.py

```diff
@@ -60,9 +62,18 @@ def __init__(self, settings_file: Path):
             "saved_state_expiry_days",
             "simulationname",
             "username",
+            "process_basic_results",
         ]
+        optional_properties = ["results_scenario_name"]
         for property_name in required_properties:
             self._read_property(property_name)
+
+        for property_name in optional_properties:
+            try:
+                self._read_property(property_name)
+            except:
```

Comment:
Een kale except! Heiden!

Reply:
Hoe zou jij dit dan oplossen? Zie ook comment hieronder

Reply:
Niet met een except (en zeker geen "kale", dan zou ik `except MissingSettingException` hebben gedaan), maar door expliciet toe te staan dat een property optioneel is. Zoiets:

    def _read_property(self, property_name, optional=False):
        """Extract <properties><string> element with the correct key attribute."""
        xpath = f"pi:properties/pi:string[@key='{property_name}']"
        elements = self._root.findall(xpath, NAMESPACES)
        if not elements and not optional:
            raise MissingSettingException(
                f"Required setting '{property_name}' is missing "
                f"under <properties> in {self.settings_file}."
            )

Dus "and not optional" in de conditie. En default is dat 'ie verplicht is.